### PR TITLE
[Clang] Remove __is_nullptr

### PR DESCRIPTION
**Pull request by philnik777**
`is_null_pointer` can be implemented very efficiently as `__is_same(__remove_cv(T), decltype(nullptr))`. Since GCC supports both of these builtins as well, libc++ has no interest in using `__is_nullptr` instead. Furthermore, I could find only a single use in the wild (https://sourcegraph.com/search?q=context:global+__is_nullptr%28+-file:clang&patternType=keyword&sm=0). Because of these reasons I don't think it's worth keeping this builtin around.

### CODE REVIEW

**Comment by cor3ntin** [comment]
It's probably reasonable to do that, however to be on the safe side, let's merge that after the Clang 19 branch. 
It will need a release note too,

Thanks!

**Review by AaronBallman — COMMENTED** [review]
I mostly support the removal but I'd feel more comfortable documenting the extension as deprecated in Clang 19, adding a release note warning users about it going away and what the replacement code would look like, and then remove `__is_nullptr()` in Clang 20. (edited to remove the need for a time machine as a prerequisite.)

That said, has anyone checked with Embarcadero and Microsoft as to whether they still need it supported for their STL implementations for historical reasons? That might delay the timeline or change opinions on the direction.

**Comment by philnik777** [comment]
> I mostly support the removal but I'd feel more comfortable documenting the extension as deprecated in Clang 19, adding a release note warning users about it going away and what the replacement code would look like, and then remove `__is_nullptr()` in Clang 20. (edited to remove the need for a time machine as a prerequisite.)
> 
> That said, has anyone checked with Embarcadero and Microsoft as to whether they still need it supported for their STL implementations for historical reasons? That might delay the timeline or change opinions on the direction.

AFAICT MSVC never had such a trait. I don't know why it's documented that way, but the documentation is almost certainly wrong.

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -117,7 +121,7 @@ Improvements to Clang's diagnostics
 - Some template related diagnostics have been improved.
 
   .. code-block:: c++
-    
+
```
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:124]: Spurious whitespace change.

File: clang/lib/Parse/ParseDeclCXX.cpp [hunk 2]
```diff
@@ -447,9 +447,9 @@ Decl *Parser::ParseLinkage(ParsingDeclSpec &DS, DeclaratorContext Context) {
 ///
 /// HLSL: Parse export function declaration.
 ///
-///      export-function-declaration: 
+///      export-function-declaration:
 ///         'export' function-declaration
-/// 
+///
```
Comment (AaronBallman) [inline clang/lib/Parse/ParseDeclCXX.cpp:452]: Spurious whitespace changes

**Review by AaronBallman — APPROVED** [review]
LGTM (modulo whitespace nits), but please make sure the 19.x branch gets a release note officially deprecating the trait and saying it will be removed in Clang 20.

**Pull request merged**